618 reduce dependencies of the package

m7pr opened the pull request:
Close #618 

Moved from `Imports` to `Suggests`
- `styler` d42dbfc7bb2bfd6d41f6e53bb9a08a0858243450 because of https://github.com/insightsengineering/teal.modules.clinical/issues/618#issuecomment-1911710827 and usage in vignettes 8a9a80e288e1eaac2f7d37c5aef7023515aa8e7a

Removed
- `magrittr` 772cf9dbfd0176e2c4b1e1dff50a7806b419ee78 because of https://github.com/insightsengineering/teal.modules.clinical/issues/618#issuecomment-1911697690
- `tidyr` e585e0a395b1cd3c3b1dacc13adb15454a1213f4 because of https://github.com/insightsengineering/teal.modules.clinical/issues/618#issuecomment-1911716602 and https://github.com/insightsengineering/teal.modules.clinical/issues/618#issuecomment-1912277877
- `rlang` f2c0bfd7f9fd229a4ab274bbfc1d3ab1ebb72513 04537f1051d541d14190307f5f15b3bb3cbd268b a94b84bba63ff6996be428a5dd86b7f4b8034009 because of https://github.com/insightsengineering/teal.modules.clinical/issues/618#issuecomment-1912313877 and https://github.com/insightsengineering/teal.modules.clinical/issues/618#issuecomment-1912321290

Melkiades reviewed (dismissed):
Lgtm! Thanks @m7pr for finding and removing these unnecessary deps ;)

m7pr commented:
@Melkiades there's gonna be a lot more in this PR. So far this is a DRAFT as this is WIP

m7pr commented:
Hey @Melkiades I would kindly like to ask for your help in this PR!
We changed the source code of 4 functions: `tm_t_events`, `tm_t_events_summary`, `tm_t_p_laboratory`, `facet_grid_formula`.
Would you be able to run examples for those functions on `main` branch and on this feature branch and let me know if you receive the same outputs/results? Thanks for the help!

m7pr commented:
There is a NOTE in the R CMD CHECK that says there is +26 packages in Imports : D our GHA does not allow to have such a NOTE in the R CMD CHECK. No idea how to merge that :)

averissimo reviewed (commented):
There's still a `library(magrittr)` on a vignette that needs to be replaced (this also generates a NOTE)

![image](https://github.com/insightsengineering/teal.modules.clinical/assets/211358/cdf094da-f0bf-4539-8a74-fc81f61c522a)

About `{styler}` usage: We might not need to use private function as `syler:::print.vertical(...)` will be dispatched via generic

Suggestions on `{diff}` below

```diff
diff --git a/vignettes/quickstart_substitute.Rmd b/vignettes/quickstart_substitute.Rmd
index 1369636c..d1c6e574 100644
--- a/vignettes/quickstart_substitute.Rmd
+++ b/vignettes/quickstart_substitute.Rmd
@@ -204,7 +204,7 @@ provided and not the expression generating the dataframe:
 the pipeline is working but not humanly readable.
 
 ```{r}
-library(magrittr)
+library(dplyr)
 
 short_iris <- head(iris)
 plot_expr <- substitute(
@@ -320,7 +320,7 @@ library(teal)
 #' @param expr (`call`)\cr or possibly understood as so.
 #'
 styled_expr <- function(expr) {
-  styler:::print.vertical(
+  print(
     styler::style_text(text = deparse(expr)),
     colored = FALSE
   )
```

m7pr commented:
Thanks for this comment @averissimo 
I had vignettes folder removed from my local find/gsub procedure and omitted this directory complitely.

Thanks!

Melkiades commented:
> Hey @Melkiades I would kindly like to ask for your help in this PR! We changed the source code of 4 functions: `tm_t_events`, `tm_t_events_summary`, `tm_t_p_laboratory`, `facet_grid_formula`. Would you be able to run examples for those functions on `main` branch and on this feature branch and let me know if you receive the same outputs/results? Thanks for the help!

`tm_t_events` when doing alphabetical sorting:
object 'AEBODSYS' not found 
 when evaluating qenv code:
anl[[AEBODSYS]] <- as.character(anl[[AEBODSYS]])

m7pr commented:
@Melkiades any way you can share code you used for testing that : ) ? thanks.
Do you see any other issues with different functions?

m7pr commented:
I was able (at least I hope) to fix `tm_events_summary` with last commit. Using below app

```r
library(teal.modules.clinical)
library(scda)
library(dplyr)
adsl <- synthetic_cdisc_dataset("latest", "adsl")
adae <- synthetic_cdisc_dataset("latest", "adae")
add_event_flags <- function(dat) {
  dat %>%
    dplyr::mutate(
      TMPFL_SER = AESER == "Y",
      TMPFL_REL = AEREL == "Y",
      TMPFL_GR5 = AETOXGR == "5",
      TMP_SMQ01 = !is.na(SMQ01NAM),
      TMP_SMQ02 = !is.na(SMQ02NAM),
      TMP_CQ01 = !is.na(CQ01NAM)
    ) %>%
    var_relabel(
      TMPFL_SER = "Serious AE",
      TMPFL_REL = "Related AE",
      TMPFL_GR5 = "Grade 5 AE",
      TMP_SMQ01 = aesi_label(dat[["SMQ01NAM"]], dat[["SMQ01SC"]]),
      TMP_SMQ02 = aesi_label(dat[["SMQ02NAM"]], dat[["SMQ02SC"]]),
      TMP_CQ01 = aesi_label(dat[["CQ01NAM"]])
    )
}
# Generating user-defined event flags.
adae <- adae %>% add_event_flags()
ae_anl_vars <- names(adae)[startsWith(names(adae), "TMPFL_")]
aesi_vars <- names(adae)[startsWith(names(adae), "TMP_")]
app <- init(
  data = cdisc_data(
    ADSL = adsl,
    ADAE = adae,
    code = '
      add_event_flags <- function(dat) {
        dat %>%
          dplyr::mutate(
            TMPFL_SER = AESER == "Y",
            TMPFL_REL = AEREL == "Y",
            TMPFL_GR5 = AETOXGR == "5",
            TMP_SMQ01 = !is.na(SMQ01NAM),
            TMP_SMQ02 = !is.na(SMQ02NAM),
            TMP_CQ01 = !is.na(CQ01NAM)
          ) %>%
          var_relabel(
            TMPFL_SER = "Serious AE",
            TMPFL_REL = "Related AE",
            TMPFL_GR5 = "Grade 5 AE",
            TMP_SMQ01 = aesi_label(dat[["SMQ01NAM"]], dat[["SMQ01SC"]]),
            TMP_SMQ02 = aesi_label(dat[["SMQ02NAM"]], dat[["SMQ02SC"]]),
            TMP_CQ01 = aesi_label(dat[["CQ01NAM"]])
          )
      }
      ADAE <- ADAE %>% add_event_flags()',
    check = TRUE
  ),
  modules = modules(
    tm_t_events_summary(
      label = "Adverse Events Summary",
      dataname = "ADAE",
      arm_var = choices_selected(
        choices = variable_choices("ADSL", c("ARM", "ARMCD")),
        selected = "ARM"
      ),
      flag_var_anl = choices_selected(
        choices = variable_choices("ADAE", ae_anl_vars),
        selected = ae_anl_vars[1],
        keep_order = TRUE,
        fixed = FALSE
      ),
      flag_var_aesi = choices_selected(
        choices = variable_choices("ADAE", aesi_vars),
        selected = aesi_vars[1],
        keep_order = TRUE,
        fixed = FALSE
      ),
      add_total = TRUE
    )
  )
)
if (interactive()) {
  shinyApp(app$ui, app$server)
}
```

m7pr commented:
Fixed `tm_t_pp_laboratory` with the latest commit.

Tested on

```r
ADSL <- tmc_ex_adsl
ADLB <- tmc_ex_adlb

app <- init(
  data = cdisc_data(
    ADSL = ADSL,
    ADLB = ADLB,
    code = "
      ADSL <- tmc_ex_adsl
      ADLB <- tmc_ex_adlb
    "
  ),
  modules = modules(
    tm_t_pp_laboratory(
      label = "Vitals",
      dataname = "ADLB",
      patient_col = "USUBJID",
      paramcd = choices_selected(
        choices = variable_choices(ADLB, "PARAMCD"),
        selected = "PARAMCD"
      ),
      param = choices_selected(
        choices = variable_choices(ADLB, "PARAM"),
        selected = "PARAM"
      ),
      timepoints = choices_selected(
        choices = variable_choices(ADLB, "ADY"),
        selected = "ADY"
      ),
      anrind = choices_selected(
        choices = variable_choices(ADLB, "ANRIND"),
        selected = "ANRIND"
      ),
      aval_var = choices_selected(
        choices = variable_choices(ADLB, "AVAL"),
        selected = "AVAL"
      ),
      avalu_var = choices_selected(
        choices = variable_choices(ADLB, "AVALU"),
        selected = "AVALU"
      )
    )
  )
)
if (interactive()) {
  shinyApp(app$ui, app$server)
}
```

m7pr commented:
The issue that you found appeared because in `substitute(expr = , env = )` we used `as.name` for parameters passed to `env` where I assumed we should be using their values and not `names`.

Melkiades commented:
Thanks, @m7pr, for solving these issues! I was playing around with the saving functionality, and I noticed two things:

1. If I save directly to pdf it truncates the table (default `tm_t_events` example):
![image](https://github.com/insightsengineering/teal.modules.clinical/assets/11279768/34f3b7b5-9cee-4f6c-b171-a92c8add872a)
Also zooming in before it does truncate it, I might be missing something I need to set but I could not find it in the download menu.

2. When adding pagination the only option is lines per page which seems to work fine, while it does automatically column wrapping w/o any way to change its values.

These are very problematic for the general user but are preceding this PR. So, this is good to go, I will open a new issue regarding this.

Melkiades reviewed (approved):
Thanks @m7pr for reducing the dependencies and fixing the errors! If you think it is necessary add a NEWS entry ;)
